Source/JavaScriptCore/runtime/FunctionPrototype.cpp

```diff
@@ -193,8 +193,19 @@ class RetrieveCallerFunctionFunctor {
             return IterationStatus::Continue;
         }
 
-        if (callee)
+        if (callee) {
+            if (callee->inherits<JSBoundFunction>() || callee->inherits<JSRemoteFunction>() || callee->type() == ProxyObjectType)
+                return IterationStatus::Continue;
+            if (callee->inherits<JSFunction>()) {
+                auto* function = jsCast<JSFunction*>(callee);
+                if (function->executable()->implementationVisibility() == ImplementationVisibility::Private)
+                    return IterationStatus::Continue;
+                if (function->isHostOrBuiltinFunction())
```

Comment:
Do we actually need this check? If we return a `function` here that is `isHostOrBuiltinFunction`, it will be rejected a bit later in `callerGetter`:

```
    if (!function || function->isHostOrBuiltinFunction())
        return JSValue::encode(jsNull());
```

I find it nicer to separate call frame iteration where we `IterationStatus::Continue` or not with vetting the final result, so maybe it's worth to have `isHostOrBuiltinFunction` only in `callerGetter`?

Reply:
That's nice. Dropped.